pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -135,6 +135,21 @@ type PrefilterPlugin interface {
 	Prefilter(pc *PluginContext, p *v1.Pod) *Status
 }
 
+// FilterPlugin is an interface for Filter plugins. These plugins are called at the
+// filter extension point for filtering out hosts that cannot run a pod.
+// This concept used to be called 'predicate' in the original scheduler.
+// These plugins should return only Success or Error in Status.code. However,
```

Comment:
It makes more sense for the Filter plugins to return `Unschedulable` code rather than an `Error` when a node cannot be used to schedule the given pod. We should add that in the comment.

Reply:
+1, also please make [this check](https://github.com/kubernetes/kubernetes/pull/78477/files#diff-c237cdd9e4cb201118ca380732d7f361R511) != Unschedulable rather than == Error.

Reply:
Oh, that check is actually correct. If should remain `== framework.Error`. That check is there to collect errors other than a `fit` error.

Reply:
Right, but a fit error is ```framework.Unschedulable```, and so to collect errors other than a fit error we should check that the error is ```!= framework.Unschedulable```, right?

Reply:
@ahg-g 
   Filter plugin should return Success, Unschedulable or other codes such as Wait, Error, and so on.

   Once it obviously returns Error, we would collect  it.

                                if !status.IsSuccess() {
                                        predicateResultLock.Lock()
                                        failedPredicateMap[nodeName] = append(failedPredicateMap[nodeName],
                                                predicates.NewFailureReason(status.Message()))
                                        if status.Code() == framework.Error {
                                                errs[status.Message()]++
                                        }
                                        predicateResultLock.Unlock()
                                        return
                                }

Reply:
What does ```Wait``` mean for a filter plugin? In your current implementation it is semantically equivalent to ```Unschedulable```, which is not correct. 

A filter plugin should either return ```Success``` or ```Unschedulable```, all other statuses should be considered equivalent to ```Error```.

Reply:
You want to treat other codes as `Error` except `Success` or `Unschedulable`.

@bsalamat 
    Do we need to enforce it or just count Error?

Reply:
Yes, I agree with @ahg-g. We should treat anything other than `Success` and `Unschedulable` as `Error` for Filter plugins.